src/zmq/zmqnotificationinterface.cpp

```diff
@@ -179,3 +181,37 @@ void CZMQNotificationInterface::NotifyTransactionLock(const CTransaction &tx)
         }
     }
 }
+
+void CZMQNotificationInterface::NotifyGovernanceVote(const CGovernanceVote &vote)
+{
+    for (std::list<CZMQAbstractNotifier*>::iterator i = notifiers.begin(); i != notifiers.end(); )
+    {
+        CZMQAbstractNotifier *notifier = *i;
```

Comment:
This is an extra un-necessary variable. You can get the value of the iterator by dereferencing, e.g. `*i`.

Reply:
What is the error message?

Reply:
At each point where notifier was replaced with the iterator, as you described.
```
zmq/zmqnotificationinterface.cpp:174:17: error: request for member ‘Shutdown’ in ‘* it.std::_List_iterator<_Tp>::operator-><CZMQAbstractNotifier*>()’, which is of pointer type 
‘CZMQAbstractNotifier*’ (maybe you meant to use ‘->’ ?)
             it->Shutdown();
                 ^
```

Reply:
Ah, the list is a list of pointers. Try dereferencing the iterator first, then calling the method.

`(*it)->Shutdown();`

Reply:
that did it 👍